### ISSUE TITLE
Agent stops working after lengthy period of laptop closed

### ISSUE THREAD

**Issue by oaksprout** [valory-xyz/olas-operate-app#222]
Reports:

```
I'm having an issue, but it could be an edge case. I started my agent at home and it made some transactions, but then closed my laptop and brought it to the cafe. I am now connected to the wifi in the cafe but the agent did not restart transacting. Here are the logs:
```
[pearl_logs (11).zip](https://github.com/user-attachments/files/16123913/pearl_logs.11.zip)
[log (1).txt](https://github.com/user-attachments/files/16123914/log.1.txt)

```
Agent was running, closed laptop, moved to new location/different wifi – agent still running but agent doesn't transact.
```
https://valory-workspace.slack.com/files/U044UPWNZ5L/F07B7M31EP7/logs.txt
[log (2).txt](https://github.com/user-attachments/files/16123921/log.2.txt)

```
I notice that after leaving the app and agent running over night, the aea_bin process has died. Have left the agent running for over an hour and there are no aea_bin processes
```
https://valory-workspace.slack.com/files/U044UPWNZ5L/F07BDRTTWRJ/image.png
[log (3).txt](https://github.com/user-attachments/files/16123935/log.3.txt)
https://valory-workspace.slack.com/files/U044UPWNZ5L/F07B4Q3CJ86/logs.txt

In summary, whatever mode the Mac goes into when closed appears to eventually kill some critical processes – namely aea_bin – and the result is that the agent appears to be working, but really has died. The automated restart does not solve this issue.

**Comment by oaksprout** [comment]
Related to #214

**Comment by oaksprout** [comment]
Fixed in rc93
